cluster-autoscaler/estimator/threshold_based_limiter.go

```diff
@@ -27,16 +27,20 @@ import (
 type thresholdBasedEstimationLimiter struct {
 	maxDuration time.Duration
 	maxNodes    int
+	limits      []BinpackingLimit
 	nodes       int
 	start       time.Time
 }
 
-func (tbel *thresholdBasedEstimationLimiter) StartEstimation([]*apiv1.Pod, cloudprovider.NodeGroup) {
+func (tbel *thresholdBasedEstimationLimiter) StartEstimation(_ []*apiv1.Pod, _ cloudprovider.NodeGroup, runtimeLimits []BinpackingLimit) {
 	tbel.start = time.Now()
 	tbel.nodes = 0
+	tbel.maxNodes = getNodeLimit(tbel.limits, 0)
+	tbel.maxNodes = getNodeLimit(runtimeLimits, tbel.maxNodes)
```

Comment:
nit: `tbel.maxNodes = getNodeLimit(append(tbel.limits, runtimeLimits...), 0)` also works AND it would allow us to drop the second argument of getNodeLimit as it would always be 0.

Reply:
We can potentially if the array underneath the `tbel.Limits` silce is big enough. But slice is a pair of (array, length), so even if we do we won't change anything in the slice itself, as all the relevant values in the array are the same and the length does not change.

Reply:
I have refactored it in 2742946b9ca375b584f7a2ef2eedd69b8a736fea